packages/flutter/lib/src/material/input_border.dart

```diff
@@ -482,14 +482,14 @@ class OutlineInputBorder extends InputBorder {
     // Draw top border from gap end to top right corner and draw top right corner.
     const double trCornerArcStart = (3 * math.pi) / 2.0;
     const double trCornerArcSweep = cornerArcSweep;
-    if (start + extent < scaledRRect.width - scaledRRect.trRadiusX) {
-      path.moveTo(scaledRRect.left + start + extent, scaledRRect.top);
+    if (start + extent < width - scaledRRect.trRadiusX) {
+      path.moveTo(start + extent, scaledRRect.top);
```

Comment:
Because `scaledRRect.width` is based on stroke width, replace it with `outer.width`.